src/Lti/LoadBalancer/UserGroupIdLtiInstanceLoadBalancer.php

```diff
@@ -0,0 +1,40 @@
+<?php declare(strict_types=1);
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2019 (original work) Open Assessment Technologies S.A.
+ */
+
+namespace App\Lti\LoadBalancer;
+
+use App\Entity\User;
+use App\Lti\Exception\IndeterminableLtiInstanceUrlException;
+
+class UserGroupIdLtiInstanceLoadBalancer extends AbstractLtiInstanceLoadBalancer
+{
+    /**
+     * @throws IndeterminableLtiInstanceUrlException
+     */
+    public function getLtiInstanceUrl(User $user): string
+    {
+        if (!$user->hasGroupId()) {
+            throw new IndeterminableLtiInstanceUrlException(
```

Comment:
What to do if the user does not have group id? Just exit with errors?

Reply:
Yes. IMHO It's better to exit instead of forwarding every user to the same instance due to missing group id.

Reply:
And just to check from my side:
since we have the `groupId` field nullable in [here](https://github.com/oat-sa/simple-roster/pull/101/files#diff-078a85fb273a48b46623a07626dab15bR33) should we really throw an exception and exit? Even if this is a `UserGroupIdLoadBalancer`?

Reply:
As I said above, it's a requirement to have group id in order to use this strategy. Otherwise it would forward all the users to the same instance, and I think it's not what we want :)

Reply:
Yep, agree. Just one moment: Depp has some users (smth in `admin_users.csv` files) that really doesn't relate to any group -> as I remember it's for an internal "testing" of teachers.
p.s.
Ofc, we can workaround it with just assigning some "generated" group for such users :)

Reply:
We'll use the school id as group id if you remember. Can you confirm that all students belong to a school?

Reply:
It was in their files. But I don't really remember if it was them or someone other having such accounts (for internal "teachers training") with a group that wasn't existent in other files -> so roster ingestion failed because of using unknow group (it was just a stub for a UAI field, that came from their side).
Smth like that, sorry, that's the only thing I remember.

Reply:
I think we can address these issues while working on the CSV adaptations. We'll need a school id for every user, otherwise the proctoring won't work

Reply:
Agreed. Thanks for the answers :+1: